src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewDataDictionaryFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class ViewDataDictionaryFactory
+    {
+        public static ViewDataDictionary CreateViewDataDictionary(IModelMetadataProvider metadataProvider)
```

Comment:
We weren't sure if there ought to be a new public ctor for this. If that's valuable, log a new issue for that, along with justification. The goal here was to not change anything that wasn't required.